nautobot/docs/development/apps/api/platform-features/uniquely-identify-objects.md

```diff
@@ -0,0 +1,60 @@
+# Uniquely Identifying a Nautobot Object
+
+In Nautobot v1.X, the slug field of many models could be used to uniquely identify a specific instance in the database. This was a convenient way to reference an object, but the slug field had some drawbacks and slugs were ultimately removed in Nautobot v2.0. As a result, new patterns will have to be adopted when trying to retrieve specific objects from the database.
+
+## Primary Keys vs. Natural Keys
+
+When designing an application for Nautobot, one of the key decisions is whether to use a primary key or a natural key to uniquely identify individual model instances. Here are some trade-offs to consider:
+
+### Primary Keys
+
+The primary key is a concrete database field and is the primary method for referencing an object or relating multiple objects in the Nautobot database. Here are some advantages of using primary keys:
+
+- **Uniqueness:** Primary keys are guaranteed to be unique.
+- **Performance:** Primary keys are always indexed in Nautobot, which makes lookups very fast.
+
+However, there are also some disadvantages to using primary keys:
+
+- **Usability**: Primary keys in Nautobot use UUIDs and can be difficult to work with directly.
+- **Portability:** Primary keys are specific to a single database and cannot be used to identify an object in multiple contexts, such as in external applications or in different databases.
+
+### Natural Keys
```

Comment:
Should we throw a doc reference to `docs/development/core/natural-keys.md` somewhere here?